### ISSUE TITLE
Exception if user does not exist

### ISSUE THREAD

**Issue by ehcpdeveloper** [giampaolo/pyftpdlib#400]
when there is no user, and that user tries to log in, this exception occurs:
```
ERROR:pyftpdlib:unhandled exception in instance <**main**.My_Handler (addr=127.0.0.1:53494, user=u'd')>
Traceback (most recent call last):
  File "/usr/lib/python2.7/asyncore.py", line 83, in read
    obj.handle_read_event()
  File "/usr/lib/python2.7/asyncore.py", line 449, in handle_read_event
    self.handle_read()
  File "/usr/local/lib/python2.7/dist-packages/pyftpdlib/ioloop.py", line 919, in handle_read
    asynchat.async_chat.handle_read(self)
  File "/usr/lib/python2.7/asynchat.py", line 165, in handle_read
    self.found_terminator()
  File "/usr/local/lib/python2.7/dist-packages/pyftpdlib/handlers.py", line 1379, in found_terminator
    self.pre_process_command(line, cmd, arg)
  File "/usr/local/lib/python2.7/dist-packages/pyftpdlib/handlers.py", line 1485, in pre_process_command
    self.process_command(cmd, arg, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pyftpdlib/handlers.py", line 1496, in process_command
    method(_args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pyftpdlib/handlers.py", line 2049, in ftp_QUIT
    msg_quit = self.authorizer.get_msg_quit(self.username)
  File "/usr/local/lib/python2.7/dist-packages/pyftpdlib/authorizers.py", line 229, in get_msg_quit
    return self.user_table[username]['msg_quit']
KeyError: u'd'
```
the problem in `get_msg_quit` function, should check if user exists , like .. `if self.user_table.has_key(username) ....`

**Comment by lurch** [comment]
What version of pyftpdlib are you using? The current code does:
```python
    def ftp_QUIT(self, line):
        """Quit the current session disconnecting the client."""
        if self.authenticated:
            msg_quit = self.authorizer.get_msg_quit(self.username)
        else:
            msg_quit = "Goodbye."
....
```
so if the user doesn't exist, I'd expect authenticated to be False?

**Comment by captincook** [comment]
I have a similar problem I need FTP accepting any user/password that's connecting. Is this possible with pyftpdlib ? can I modify authenticated to return always true ?

**Comment by ehcpdeveloper** [comment]
Sure you can. In similar way that I did. download ehcp.net/ehcp_yeni.tgz and check ftp_server2.py code. Ehcp currently uses pyftpdlib, successfully. 
I overcomed above problem, by encapsulating ftp_QUIT function, in the derived class.

**Comment by captincook** [comment]
Thanks. I tried to use http://pythonhosted.org/pyftpdlib/tutorial.html#event-callbacks example and subclass on_login method. I installed everything (I'm on Linux) but get an error:

`from pyftpdlib.servers import DummyAuthorizer
ImportError: cannot import name DummyAuthorizer`

Was DummyAuthorizer moved/ removed ?

**Comment by lurch** [comment]
https://github.com/giampaolo/pyftpdlib/blob/master/docs/tutorial.rst#event-callbacks has the correct code.

@giampaolo looks like the docs on http://pythonhosted.org/pyftpdlib/ are out of date? (or haven't been correctly generated, since it claims to be "pyftpdlib 1.5.0 documentation")

**Comment by giampaolo** [comment]
To my understanding this problem occurs if user sends QUIT after USER without having sent PASS (so auth didn't occur). I should have fixed this in 4dea0a7b46cdd13821e15b48f599d70e3bf93b3e.
